sources/packages/backend/apps/api/src/route-controllers/application/application.aest.controller.ts

```diff
@@ -183,73 +203,111 @@ export class ApplicationAESTController extends BaseController {
 
   /**
    * Get details for an application at 'InProgress' status.
-   * @param applicationId application id.
+   * @param applicationId current application id.
    * @returns application details.
    */
   @Get(":applicationId/in-progress")
   @ApiNotFoundResponse({
-    description: "Application id not found.",
+    description:
+      "Application id not found or current application for provided parent application not found.",
   })
   @ApiUnprocessableEntityResponse({
     description: `Application not in ${ApplicationStatus.InProgress} status.`,
   })
   async getInProgressApplicationDetails(
     @Param("applicationId", ParseIntPipe) applicationId: number,
   ): Promise<InProgressApplicationDetailsAPIOutDTO> {
+    const currentApplicationId =
+      await this.applicationService.getApplicationIdByParentApplicationId(
+        applicationId,
+      );
+    if (!currentApplicationId) {
+      throw new NotFoundException(
+        `Current application for application ${applicationId} was not found.`,
+      );
+    }
     return this.applicationControllerService.getInProgressApplicationDetails(
-      applicationId,
+      currentApplicationId,
     );
   }
 
   /**
    * Get status of all requests and confirmations in student application (Exception, PIR and COE).
-   * @param applicationId application id.
+   * @param applicationId current application id.
    * @returns application progress details.
    */
   @ApiNotFoundResponse({
-    description: "Application not found.",
+    description:
+      "Application not found or current application for provided parent application not found.",
   })
   @Get(":applicationId/progress-details")
   async getApplicationProgressDetails(
     @Param("applicationId", ParseIntPipe) applicationId: number,
   ): Promise<ApplicationProgressDetailsAPIOutDTO> {
+    const currentApplicationId =
+      await this.applicationService.getApplicationIdByParentApplicationId(
+        applicationId,
+      );
+    if (!currentApplicationId) {
+      throw new NotFoundException(
+        `Current application for application ${applicationId} was not found.`,
+      );
+    }
     return this.applicationControllerService.getApplicationProgressDetails(
-      applicationId,
+      currentApplicationId,
```

Comment:
Same set of comments

TODO:
Not found error message correction.
Call applicationId as parentApplicationId